### ISSUE TITLE
asi bug is not fixed

### ISSUE THREAD

**Issue by ChALkeR** [nodejs/amaro#24]
ref: https://github.com/swc-project/swc/issues/9331

that fix is not complete

```ts
const x = 10 as number
(1)
```

is
```js
"use strict";
const x = 10;
(1);
```
in typescript playground

But here it transforms into
```js
const x = 10      
(1)
```

causing `TypeError: 10 is not a function`

While the snippet above causes a visible error, this one below will cause unexpected code to run silentely:
```ts
const y = () => () => console.log('unexpected')
const x = y as Function
(() => console.log('expected'))()
```

Seems to be an upstream bug?

**Comment by marco-ippolito** [comment]
cc @kdy1

**Comment by kdy1** [comment]
I filed https://github.com/swc-project/swc/issues/9355

**Comment by magic-akari** [comment]
This is more challenging than other ASI issues, because the 'as' expression isn't a standalone statement. 
It must be fixed when it appears at the end of a statement, but it can also be found within an expression.

**Comment by magic-akari** [comment]
fixed in https://github.com/swc-project/swc/pull/9358

**Comment by kdy1** [comment]
https://github.com/swc-project/swc/pull/9358 is merged, and I published `1.7.3`
